### DISCUSSION TITLE
Put.io integration?

### DISCUSSION THREAD

**Discussion by itsToggle** [itsToggle/plex_debrid#47] [Announcements]
Alright so ive been asked if a put.io integration is possible.

It is, but put.io doesnt work in the same way as other debrid services - it doesnt allow you to check wether or not a torrent is cached before starting a download. I can integrate put.io by simply downloading the best release that was found, regardless of wether its cached or not.

Put.io does not offer a free trial for me to test out any implementation. If this is something that more than one person is interested in, ill think about paying for a subscription. Let me know if you guys are interested and/or got any ideas on this.

**Comment by bitsper2nd** [comment]
Put.io has instructions for using it with Plex. You can [find it here](http://help.put.io/en/articles/3480094-plex-rclone). If you are interested in working on this, I can transfer money on paypal for the [one month prepaid option](https://put.io/plan?code=casual&discount_percent=0&payment_period=30&payment_type=credit-card&plan_type=onetime&sub_plan_code=casual_monthly_onetime).

**Reply by itsToggle** [reply]
Thanks! But no need - I just found out that they offer a one day membership for a dollar, which I do have spare :) no idea how I missed this.

I'll commit the put.io integration tomorrow :)

**Reply by itsToggle** [reply]
alright, put.io is integrated. Again, it wont be able to check wether a release is cached or not. I tried to create a cache check manually, but since I would need to remove each transfer and the assoiciated files for each checked release, ive decided against implementing the cache check. The put.io service will simply download the best release that was found - regardless of cache stauts.
